tests/ignite/metrics/test_precision.py

```diff
@@ -47,168 +47,145 @@ def test_binary_wrong_inputs():
         pr.update((torch.randint(0, 2, size=(10,)).long(), torch.randint(0, 2, size=(10, 5, 6)).long()))
 
 
-def test_binary_input_N():
+@pytest.mark.parametrize("average", [False, True])
+def test_binary_input_N(average):
     # Binary accuracy on input of shape (N, 1) or (N, )
 
-    def _test(average):
-        pr = Precision(average=average)
-        y_pred = torch.randint(0, 2, size=(10,))
-        y = torch.randint(0, 2, size=(10,)).long()
-        pr.update((y_pred, y))
-        np_y = y.numpy().ravel()
-        np_y_pred = y_pred.numpy().ravel()
-        assert pr._type == "binary"
-        assert isinstance(pr.compute(), float if average else torch.Tensor)
-        pr_compute = pr.compute() if average else pr.compute().numpy()
-        assert precision_score(np_y, np_y_pred, average="binary") == pytest.approx(pr_compute)
+    pr = Precision(average=average)
 
+    def _test(y_pred, y, batch_size):
         pr.reset()
-        y_pred = torch.randint(0, 2, size=(10,))
-        y = torch.randint(0, 2, size=(10,)).long()
         pr.update((y_pred, y))
-        np_y = y.numpy().ravel()
-        np_y_pred = y_pred.numpy().ravel()
-        assert pr._type == "binary"
-        assert isinstance(pr.compute(), float if average else torch.Tensor)
-        pr_compute = pr.compute() if average else pr.compute().numpy()
-        assert precision_score(np_y, np_y_pred, average="binary") == pytest.approx(pr_compute)
 
-        pr.reset()
-        y_pred = torch.Tensor([0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.51])
-        y_pred = torch.round(y_pred)
-        y = torch.randint(0, 2, size=(10,)).long()
-        pr.update((y_pred, y))
         np_y = y.numpy().ravel()
         np_y_pred = y_pred.numpy().ravel()
+
+        if batch_size > 1:
+            n_iters = y.shape[0] // batch_size + 1
+            for i in range(n_iters):
+                idx = i * batch_size
+                pr.update((y_pred[idx : idx + batch_size], y[idx : idx + batch_size]))
+
         assert pr._type == "binary"
         assert isinstance(pr.compute(), float if average else torch.Tensor)
         pr_compute = pr.compute() if average else pr.compute().numpy()
         assert precision_score(np_y, np_y_pred, average="binary") == pytest.approx(pr_compute)
 
-        # Batched Updates
-        pr.reset()
-        y_pred = torch.randint(0, 2, size=(100,))
-        y = torch.randint(0, 2, size=(100,)).long()
+    def get_test_cases():
 
-        batch_size = 16
-        n_iters = y.shape[0] // batch_size + 1
+        test_cases = [
+            (torch.randint(0, 2, size=(100,)), torch.randint(0, 2, size=(100,)), 1),
+            (torch.randint(0, 2, size=(100, 1)), torch.randint(0, 2, size=(100, 1)), 1),
+            (torch.randint(0, 2, size=(200,)), torch.randint(0, 2, size=(200,)), 1),
+            (torch.randint(0, 2, size=(200, 1)), torch.randint(0, 2, size=(200, 1)), 1),
```

Comment:
Let's keep size as 10 <- 100 and remove test cases with 200. Let's keep 2 test cases instead of 4 for each group